redist/org.goldendict.desktop.appdata.xml

```diff
@@ -34,6 +34,6 @@
   </screenshots>
   <url type="homepage">http://goldendict.org</url>
   <update_contact>https://github.com/goldendict/goldendict</update_contact>
-  <launchable type="desktop-id">goldendict.desktop</launchable>
+  <launchable type="desktop-id">org.goldendict.desktop.desktop</launchable>
```

Comment:
Why duplicate the "desktop" word?

Reply:
oh that was a mistake... no clue why the substitution gives another

Reply:
This word is duplicated in the corresponding filename too, so I assumed it was intentional for some reason.

Reply:
Oh I know what do you mean. That's because the <id> of the metainfo file has a . desktop suffix. That's not recommended but we follow the upstream's (your) name

Reply:
If you mean `<id>org.goldendict.desktop</id>` in *goldendict.appdata.xml*, then I think the suffix can be removed there, unless it is going to cause backward compatibility issues for users that care about this ID... 

Interestingly, most KDE desktop file names in my _/usr/share/applications/_ follow the reverse-DNS scheme, as well as most *GNOME* apps, but almost no *Xfce* apps do. And not many independent apps follow it either.

Reply:
> I think the suffix can be removed there, unless it is going to cause backward compatibility issues for users that care about this ID...

If you do agree it can be changed, the [recommended way ](https://docs.flatpak.org/en/latest/conventions.html#application-ids) is:

> For some practical examples of bad IDs
> 
>     org.example.desktop
> 
> This is a bad ID because the Appstream standard for legacy reasons treats IDs ending with .desktop as a special case causing inconsistency. For this same reason, .Desktop suffixes should not be used for newly named applications. Don’t hesitate to repeat the application name even if it already is part of the domain name section of the identifier (eg. org.example.Example).
> 

which means `org.goldendict.GoldenDict`

> Interestingly, most KDE desktop file names in my /usr/share/applications/ follow the reverse-DNS scheme, as well as most GNOME apps, but almost no Xfce apps do. And not many independent apps follow it either.

My guess is the freedesktop standard is written by GNOME and KDE...

Reply:
Yes, `org.goldendict.GoldenDict` seems to conform to the specification.

I don't know if it can be changed without backward compatibility issues. Do you know what this file is used for by which applications and whether its ID/name stability matters?

Reply:
You need to use the provides tag with an id to mention the old app-id of the app before the migration

Reply:
The metainfo data is used for the store listing. The .desktop file is used for launcher listing. The only chance it will cause a backward compatibility issue is when the packager hard-code the filename when copying the file to the corresponding locations. For [archlinux](https://github.com/archlinux/svntogit-community/blob/packages/goldendict/trunk/PKGBUILD) they don't do so. But for [Fedora](https://src.fedoraproject.org/rpms/goldendict//blob/rawhide/f/goldendict.spec#_86), they may need some changes

Reply:
OK @bilelmoussaoui gives a better solution

Reply:
@bilelmoussaoui for the packaging part I guess they (e.g. Fedora) still have to make some changes, am I right?

Reply:
> @bilelmoussaoui for the packaging part I guess they (e.g. Fedora) still have to make some changes, am I right?

That's a packager's job. It shouldn't be problematic for distro packagers at all.

Reply:
> Why duplicate the "desktop" word?
Per the spec, the launchable tag contains a way  to tell the software center how to open your application. Which in your case is by using the desktop file you install with the app that's named `org.goldendict.desktop.desktop`

Reply:
On a side note, I'd risk saying that `org.goldendict.desktop` sounds stupid is just a mine of issues for all the tooling assuming `.desktop` suffix means a desktop launcher. `org.goldendict.GoldenDict` would be safer and IMHO better looking.